### ISSUE TITLE
Client-Configurable Padding Character

### ISSUE THREAD

**Issue by coip** [ianlopshire/go-fixedwidth#36]
As a consumer of this package, I have a need for generating fixedwidth -encoded output that's padded with `0`s, so that I can integrate with a system which treats spaces as significant characters.

**Comment by coip** [comment]
at first glance, looks to be wired in here currently: 

https://github.com/ianlopshire/go-fixedwidth/blob/7cd39dd347dce351d85c0e8bdf930c3927f5b6b9/encode.go#L161

seems like it'd be straightforward, could add a pkg `var padChar` to be used in the above ref?

**Comment by coip** [comment]
was initially looking at something like:
```go
type Encoder struct {
	w              *bufio.Writer
	lineTerminator []byte
	padchar        rune
}
```

vs the `pkg` owning the **padchar**, but ...

https://github.com/ianlopshire/go-fixedwidth/blob/7cd39dd347dce351d85c0e8bdf930c3927f5b6b9/encode.go#L159

does make the `encoder` owning the **padchar** more complicated than I thought at first.

Open to any feedback, suggestions.

-----

Have forked, will experiment and circle back if some solutions reached.

**Comment by ianlopshire** [comment]
@coip, This is definitely a feature I would like the library to support, but I'm unsure just what that support should look like.

I think configurable formatting on the struct field level (see #28) would support this use-case – though it may be a bit verbose.

The struct encoder issue has been discussed in #32. The package owning the encoders allows for some optimizations that I don't want to lose. The ideal solution is not clear at this point.

Feel free to experiment and see where it goes. I look forward to seeing how you solve this.

**Comment by coip** [comment]
the tag would be an elegant solution, though your point on the ` ` case raises a good point... 🤔

**Comment by coip** [comment]
shouldve provided a rough example to qualify what I meant.
high level, we can do what we need w/ struct granular padchars.

eg, (my crude understanding of our granularity needs at least, for sake of provenance)

### encodedfile.txt
```
header
datatype1[a]
datatype1[b]
datatype1[c]
...
datatype2[a]
datatype2[b]
...
footer
```

### where padchars might need to be encoded ~as

- header: { **padchar**: `0` }

- datatypes:
    - 1: { **padchar**: `1` }
    - 2: { **padchar**: `2` }

- footer: { **padchar**: `@` }

**Comment by ianlopshire** [comment]
closed by #37
